Documentation/ceph-best-practices.md

```diff
@@ -0,0 +1,516 @@
+* Configure Ceph using the central configuration database when possible. This allows for more
+  runtime configuration flexibility. Do this using `ceph config set` commands from Rook's toolbox.
+  Only use Rook’s provided ceph.conf override ConfigMap when it is required.
+
+
+## Limiting Ceph to Specific Nodes
+
+One of the more common setups you may want for your Rook-Ceph cluster is to limit Ceph to a specific
+set of nodes. Even for co-located use cases, you could have valid reasons why you must not (or do
+not want to) use some nodes for storage. This is applicable for both co-located and disaggregated
+use cases. To limit Ceph to specific nodes, we can label Kubernetes nodes and configure Rook to have
+Affinity (as a hard preference) for our label.
+
+For our example, let's label the desired storage nodes with `storage-node=true`. To run Rook and
+Ceph daemons on labeled nodes, we will configure Affinities for this label in both the Rook
+Operator manifest (`operator.yaml`) and the Ceph cluster manifest (`cluster.yaml`).
+
+For Rook daemons and the CSI driver daemons, adjust the Operator manifest. Note that the CSI
+Provisioner is best started on the same nodes as other Ceph daemons.
+
+**`operator.yaml`:**
+```yaml
+CSI_PROVISIONER_NODE_AFFINITY: “storage-node=true”
+AGENT_NODE_AFFINITY: “storage-node=true”
+DISCOVER_AGENT_NODE_AFFINITY: “storage-node=true”
+```
+
+For `all` Ceph daemons, we must add Affinity for storage nodes in `cluster.yaml`.
+
+**`cluster.yaml`:**
+```yaml
+placement:
+  all:
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+          - matchExpressions:
+              - key: storage-node
+                operator: In
+                values:
+                  - "true"
+```
+
+## Segregating Ceph from User Applications
+
+You could also have reason to totally separate Rook and Ceph nodes from application nodes. This
+falls under the disaggregated use-case, and it is a more traditional way to deploy storage. In this
+case, we still need to follow the best practices described in the section above to
+[Limit Ceph to Specific Nodes](#limiting-ceph-to-specific-nodes), and we also need some additional
+settings.
+
+To segregate Ceph from user applications, we will also label all non-storage nodes with
+`storage-node=false`. The CSI plugin pods must run where user application Pods run and not where
+Rook or Ceph Pods are run. Add a CSI plugin Affinity for all non-storage nodes in the Rook
+operator configuration.
+
+**`operator.yaml`:**
+```yaml
+CSI_PLUGIN_NODE_AFFINITY: "storage-node=false"
+```
+
+In addition to that, we will set Kubernetes node Taints and configure Rook Tolerations. For example,
+Taint the storage nodes with `storage-node=true:NoSchedule`, and then add the Tolerations shown
+below to the Rook operator in `operator.yaml`.
+
+**`operator.yaml`:**
+```yaml
+AGENT_TOLERATIONS: |
```

Comment:
No need for the agent toleration since it's not recommended to run the flex driver anymore